### ISSUE TITLE
Provide a way to use environment variables and config files

### ISSUE THREAD

**Issue by notaphplover** [robohydra/robohydra#35]
Hi :), I've been trying to start the robohydra server providing a config file and a port as an option:

`
./node_modules/robohydra/bin/robohydra.js robohydra.config.json --port=${ROBOHYDRA_PORT_INNER}
`

The robohydra.config.json file is the following one:

`
{
  "plugins": [],
  "pluginLoadPaths": ["./test/services"]
}
`

Robohydra server is started using the port 3000 (the default one) instead of using the port provided in the option.

We are interested in using env variables because our app is allocated in a docker container and this approach would be simple and clean.

I understand that setting env variables in the config file is not a good idea, so I wonder if it could be possible that options provided to the cli would have priority over the options parsed in the config file.

Thank you for your time and your awesome library :)

**Comment by emanchado** [comment]
Hm, I don't think that's a problem in RoboHydra. It shouldn't matter that the value comes from an environment variable (indeed, RoboHydra doesn't even know that), and command-line parameters should indeed take precedence over anything else.

I tried this and worked as expected:
```
$ cat >robohydra.config.json
{ "plugins": [], "pluginLoadPaths": ["./test/services"] }
$ ./bin/robohydra.js robohydra.config.json --port 3001
RoboHydra ready on port 3001 - Admin URL: http://localhost:3001/robohydra-admin
^C
$ 
```

Maybe the variable somehow isn't being passed around, or it doesn't have the value you though?

**Comment by emanchado** [comment]
And thanks for the report! :-D

**Comment by notaphplover** [comment]
Hi @emanchado, sorry for the inconvenience, it was my mistake... :(.

My script was 

`
./node_modules/robohydra/bin/robohydra.js robohydra.config.json --port=${ROBOHYDRA_PORT_INNER}
`

instead of

`
./node_modules/robohydra/bin/robohydra.js robohydra.config.json --port ${ROBOHYDRA_PORT_INNER}
`

Thank you again 😃

**Comment by emanchado** [comment]
I'll leave this open for now, because it's pretty irritating that the first line doesn't throw an error. I have just checked and I'm using an _ancient_ version of `commander`, the command-line parser.

I'll have a look and if there's an easy fix I'll make it behave in a less confusing way (either making the first line just work, or at least throw an error).

**Comment by notaphplover** [comment]
Ok. I've been testing a bit. Commander v1.3.2 seems to work fine because the version 1.0.2 of the library adds support for this. Your tests passed after upgrading to this version. Maybe some extra dependencies should be updated, but we can talk about that in another issue, if you want :).

**Comment by emanchado** [comment]
Bah, I'm having problems installing packages right now. I may have to update something in my system, but I don't feel like having a look right now :-/ As this is not really urgent or anything, I might leave it hanging for some weeks or months :sweat_smile: 

Just curious: why were you testing Commander 1.3.2? There's a 2.x version already...

**Comment by notaphplover** [comment]
Hi @emanchado :), it's ok, robohydra is suposed to be used in develop environments, so, as you say, it's not urgent. I proposed 1.3.2 because when I want to update some library I try to do it step by step (it uses to be less frustrating XD) Of course 2.x is the best approach, but I did not test if you can just update your dependency or you have to change something due to the mayor version (I was a bit lazy, if you want I could test it for you).

**Comment by emanchado** [comment]
Ok, I just tried again. It seems that the problem is Buster, the module I'm using for tests. It has been abandoned for a long time and one of its dependencies is failing to build or whatever. Not sure about the details, but it seems that the solution is to migrate the tests to a newer, better supported library.

That's going to take a while, though, but at least the basics seem to work well (the simple integration tests I wrote all pass) with an upgraded Commander (2.19). Whenever I have the time and the inclination, I'll rewrite the tests, commit the changes, and close this. I don't think I'll release a new RoboHydra version just to upgrade Commander, though. It doesn't seem to be such a high-priority problem.
